File: library/src/main/java/org/fbme/lib/iec61499/descriptors/SocketType.java
package org.fbme.lib.iec61499.descriptors;


import org.fbme.lib.iec61499.declarations.AdapterTypeDeclaration;
import org.fbme.lib.common.Declaration;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.Collections;
import java.util.List;

public class SocketType implements FBTypeDescriptor {

    private final AdapterTypeDeclaration myDeclaration;

    public SocketType(AdapterTypeDeclaration declaration) {
        myDeclaration = declaration;
    }

    @Nullable
    @Override
    public Declaration getDeclaration() {
        return myDeclaration;
    }

    @Override
    public @NotNull String getTypeName() {
        return myDeclaration.getName();
    }

    @Override
    public @NotNull List<FBPortDescriptor> getEventInputPorts() {
        return FBTypeDescriptorUtils.getEventInputPorts(myDeclaration);
    }

    @Override
    public @NotNull List<FBPortDescriptor> getEventOutputPorts() {
        return FBTypeDescriptorUtils.getEventOutputPorts(myDeclaration);
    }

    @Override
    public @NotNull List<FBPortDescriptor> getDataInputPorts() {
        return FBTypeDescriptorUtils.getDataInputPorts(myDeclaration);
    }

    @Override
    public @NotNull List<FBPortDescriptor> getDataOutputPorts() {
        return FBTypeDescriptorUtils.getDataOutputPorts(myDeclaration);
    }

    @Override
    public @NotNull List<FBPortDescriptor> getSocketPorts() {
        return Collections.emptyList();
    }

    @Override
    public @NotNull List<FBPortDescriptor> getPlugPorts() {
        return Collections.emptyList();
    }

    @Override
    public @NotNull List<Integer> getAssociatedVariablesForInputEvent(int eventNumber) {
        return FBTypeDescriptorUtils.getAssociatedVariablesForInputEvent(myDeclaration, eventNumber);
    }

    @Override
    public @NotNull List<Integer> getAssociatedVariablesForOutputEvent(int eventNumber) {
        return FBTypeDescriptorUtils.getAssociatedVariablesForOutputEvent(myDeclaration, eventNumber);
    }
}

File: code/library/src/main/kotlin/org/fbme/lib/iec61499/descriptors/SocketType.kt
package org.fbme.lib.iec61499.descriptors

import org.fbme.lib.common.Declaration
import org.fbme.lib.iec61499.declarations.AdapterTypeDeclaration

class SocketType(private val myDeclaration: AdapterTypeDeclaration) : FBTypeDescriptor {
    override val declaration: Declaration
        get() = myDeclaration
    override val typeName: String
        get() = myDeclaration.name
    override val eventInputPorts: List<FBPortDescriptor>
        get() = FBTypeDescriptorUtils.getEventInputPorts(myDeclaration)
    override val eventOutputPorts: List<FBPortDescriptor>
        get() = FBTypeDescriptorUtils.getEventOutputPorts(myDeclaration)
    override val dataInputPorts: List<FBPortDescriptor>
        get() = FBTypeDescriptorUtils.getDataInputPorts(myDeclaration)
    override val dataOutputPorts: List<FBPortDescriptor>
        get() = FBTypeDescriptorUtils.getDataOutputPorts(myDeclaration)
    override val socketPorts: List<FBPortDescriptor>
        get() = emptyList()
    override val plugPorts: List<FBPortDescriptor>
        get() = emptyList()

    override fun getAssociatedVariablesForInputEvent(eventNumber: Int): List<Int> {
        return FBTypeDescriptorUtils.getAssociatedVariablesForInputEvent(myDeclaration, eventNumber)
    }

    override fun getAssociatedVariablesForOutputEvent(eventNumber: Int): List<Int> {
        return FBTypeDescriptorUtils.getAssociatedVariablesForOutputEvent(myDeclaration, eventNumber)
    }
}
